### PR TITLE
Enhancement on Yesno()

### PR DESCRIPTION
**Pull request by Liujingfang1**

### CODE REVIEW

File: cmd/kubebuilder/util/stdin.go [hunk 1]
```diff
@@ -18,21 +18,26 @@ package util
 
 import (
 	"bufio"
+	"fmt"
 	"log"
 	"os"
 	"strings"
 )
 
+var reader = bufio.NewReader(os.Stdin)
```
Comment (droot) [inline cmd/kubebuilder/util/stdin.go]: This should be local to the `Yesno` function or taken as input in `Yesno`

Comment (mengqiy) [reply cmd/kubebuilder/util/stdin.go]: > local to the Yesno function

This won't work.

> taken as input in Yesno

This is viable.

**Comment by mengqiy** [comment]
This is fixing an issue that blocks me in testing.

File: cmd/kubebuilder/util/stdin.go (hunk 1 above)
Comment (Liujingfang1) [reply cmd/kubebuilder/util/stdin.go]: We can call `NewReader` inside api command. As long as we only create one reader, it will be ok.

**Comment by Liujingfang1** [comment]
Updated the change. I've tested it by
`cat input | kubebuilder create api --group a --version v1 --kind O`
PTAL

**Pull request merged**